File: Python/Two Pointer Approach/Separate Black and White Balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        totalswap = 0
        white =0
        for i in range(len(s)):
            if s[i]=='0':
                totalswap+=i-white
                white+=1
            
        return totalswap
        

File: CPP/Two - Pointer Approach/Separate Black and White Balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        long long totalswap=0;
        int whitepos = 0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='0'){
                totalswap+=i-whitepos;
                whitepos++;
            }
        }
        return totalswap;
    }
};
